src/pip/_internal/__init__.py

```diff
@@ -1,5 +1,6 @@
 from typing import List, Optional
 
+import pip._internal.utils._log  # noqa
```

Comment:
I don't really like having an import do important stuff like this. It's too easy for someone to think it's not needed and delete it. Could we not have an init_logging() function and call that in `main()`?

Reply:
I agree, but I think `main()` is too late. It must be done before any module that calls `getLogger()` at import time is imported, which is ~all of them.

Alternately, our custom `getLogger` could *call* `setLoggerClass`, in which case it would not be purely a type hint, but an actual function that does something.

Reply:
I don’t think there is any logger call before `main()` though? The only thing done before it is a `sys.stderr.write()` call if a legacy wrapper is invoked.

Reply:
`utils.logging` itself imports other modules, which call `logging.getLogger()`, so those modules will not get the verbose logger. It must be in a module that has no other pip._internals imports, but I can definitely make it work there.

Since we need to call the custom getLogger anyway just to get type annotations (until/unless someone who knows mypy better than me can figure out how to avoid it), what do you think about calling setLoggerClass there? Then the import-time init wouldn't be needed.

Reply:
I’m still hoping we could find a way to get rid of that wrapper (don’t need to happen in this PR) so let’s not put things there.

Reply:
Gotcha. Added `_log.init_logging()` (not in `logging` since it imports other utils which call getLogger) and comments about where/when/why.